src/allotropy/parsers/moldev_softmax_pro/softmax_pro_parser.py

```diff
@@ -1,17 +1,266 @@
+    PlateReaderDocumentItem,
+    SampleDocument,
+    ScanPositionSettingPlateReader,
+    UltravioletAbsorbancePointDetectionDeviceControlAggregateDocument,
+    UltravioletAbsorbancePointDetectionDeviceControlDocumentItem,
+    UltravioletAbsorbancePointDetectionMeasurementDocumentItems,
+)
+from allotropy.allotrope.models.shared.definitions.custom import (
+    TQuantityValueDegreeCelsius,
+    TQuantityValueMilliAbsorbanceUnit,
+    TQuantityValueNanometer,
+    TQuantityValueNumber,
+    TRelativeFluorescenceUnit,
+    TRelativeLightUnit,
+)
+from allotropy.constants import ASM_CONVERTER_NAME, ASM_CONVERTER_VERSION
+from allotropy.exceptions import (
+    AllotropeConversionError,
+)
 from allotropy.named_file_contents import NamedFileContents
 from allotropy.parsers.lines_reader import CsvReader, read_to_lines
-from allotropy.parsers.moldev_softmax_pro.softmax_pro_structure import Data
+from allotropy.parsers.moldev_softmax_pro.softmax_pro_structure import (
+    Data,
+    PlateBlock,
+)
+from allotropy.parsers.utils.values import (
+    assert_not_none,
+)
 from allotropy.parsers.vendor_parser import VendorParser
 
+EPOCH = "1970-01-01T00:00:00-00:00"
+
 
 class SoftmaxproParser(VendorParser):
-    def to_allotrope(self, named_file_contents: NamedFileContents) -> Any:
+    def to_allotrope(self, named_file_contents: NamedFileContents) -> Model:
         lines = read_to_lines(named_file_contents.contents, encoding=None)
         reader = CsvReader(lines)
         data = Data.create(reader)
-        return self._get_model(data)
+        return self._get_model(named_file_contents.original_file_name, data)
+
+    def _get_model(self, file_name: str, data: Data) -> Model:
+        return Model(
+            field_asm_manifest="http://purl.allotrope.org/json-schemas/adm/plate-reader/BENCHLING/2023/09/plate-reader.schema",
+            plate_reader_aggregate_document=PlateReaderAggregateDocument(
+                device_system_document=DeviceSystemDocument(
+                    device_identifier="null",
+                    model_number="null",
+                ),
+                data_system_document=DataSystemDocument(
+                    file_name=file_name,
+                    software_name="SoftMax Pro",
+                    software_version=None,
+                    ASM_converter_name=ASM_CONVERTER_NAME,
+                    ASM_converter_version=ASM_CONVERTER_VERSION,
+                ),
+                plate_reader_document=[
+                    self._get_plate_reader_document_item(plate_block, position)
+                    for plate_block in data.get_plate_block()
+                    for position in plate_block.iter_wells()
+                ],
+                calculated_data_aggregate_document=None,
+            ),
+        )
+
+    def _get_plate_reader_document_item(
+        self, plate_block: PlateBlock, position: str
+    ) -> PlateReaderDocumentItem:
+        if plate_block.plate_block_type == "Absorbance":
```

Comment:
extract `plate_block.plate_block_type` to a local var please -- it's used 3X

(If we could require Python >=3.10, we could use `case`, but we have to support 3.9 for now)